### PR TITLE
Support different bool string representations

### PR DESCRIPTION
**Pull request by lzakharov**
Add support for different bool string representations, mentioned in the [documentation](https://www.postgresql.org/docs/11/datatype-boolean.html).

### CODE REVIEW

**Comment by jackc** [comment]
PostgreSQL does allow multiple formats for bools on *input*, but AFAIK it only ever uses `t`/`f` on output. I don't see how this code would be called. Do you have an example of how it would be used?

**Comment by lzakharov** [comment]
It's used in binary encoding, where the value is first scanned in text format: https://github.com/jackc/pgx/blob/6defa2a607fd64cb5f792dfef0cf17476bfe7d57/pgtype/pgtype.go#L1453-L1467

Original problem described in the linked issue.

File: pgtype/bool.go [hunk 1]
```diff
@@ -309,9 +316,28 @@ func (scanPlanTextAnyToBoolScanner) Scan(src []byte, dst any) error {
...
+
+	return s.ScanBool(Bool{Bool: v, Valid: true})
+}
+
+// https://www.postgresql.org/docs/11/datatype-boolean.html
+func planTextToBool(src []byte) (bool, error) {
+	s := string(bytes.ToLower(bytes.TrimSpace(src)))
+
+	switch {
+	case strings.HasPrefix("true", s), strings.HasPrefix("yes", s), s == "on", s == "1":
```
Comment (drakkan) [inline pgtype/bool.go:336]: This will match `t`, `tr`, `tru`, `true`, is this by purpose?

Comment (lzakharov) [reply pgtype/bool.go:336]: From docs:
> Unique prefixes of these strings are also accepted

File: pgtype/bool.go [hunk 2]
```diff
@@ -309,9 +316,28 @@ func (scanPlanTextAnyToBoolScanner) Scan(src []byte, dst any) error {
...
+	if err != nil {
+		return err
+	}
+
+	return s.ScanBool(Bool{Bool: v, Valid: true})
+}
+
+// https://www.postgresql.org/docs/11/datatype-boolean.html
+func planTextToBool(src []byte) (bool, error) {
+	s := string(bytes.ToLower(bytes.TrimSpace(src)))
```
[the remark this answers is not in the document]
Comment (lzakharov) [reply pgtype/bool.go:333]: Agree, but this conversion helps to avoid `[]byte` around the string constants.

**Pull request merged**